library/src/scala/quoted/Quotes.scala

```diff
@@ -4054,6 +4013,79 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
 
     end TreeMap
 
+    /** Type class used in `show` methods to provide customizable `String` representations */
+    trait Printer[T]:
+      /** Show the arguments as a `String` */
+      def show(x: T): String
+    end Printer
+
+    /** Default pinter for `Tree` used when calling `tree.show` */
+    given TreePrinter: Printer[Tree] = Printer.TreeCode
+
+    /** Default pinter for `TypeRepr` used when calling `tpe.show` */
+    given TypeReprPrinter: Printer[TypeRepr] = Printer.TypeReprCode
+
+    /** Default pinter for `Symbol` used when calling `sym.show` */
+    given SymbolPrinter: Printer[Symbol] = Printer.SymbolFullName
+
+    /** Default pinter for `Flags` used when calling `flags.show` */
+    given FlagsPrinter: Printer[Flags] = Printer.FlagsCombination
+
+    /** Module object of `type Printer`.
+     *  Contains custom printers such as `TreeCode`, `TreeAnsiCode`, `TreeCases`, `TypeReprCode`, ..., `SymbolFullName` and `FlagsCombination`.
+     */
+    val Printer: PrinterModule
+
+    /** Methods of the module object `val Printer` */
+    trait PrinterModule { self: Printer.type =>
+      /** Prints fully elaborated vesion of the source code. */
+      def TreeCode: Printer[Tree]
+
+      /** Prints fully elaborated vesion of the source code.
+       *  Same as `TreeCode` but does not print full package prefixes.
+       */
+      def TreeShortCode: Printer[Tree]
+
+      /** Prints fully elaborated vesion of the source code using ANSI colors. */
+      def TreeAnsiCode: Printer[Tree]
+
+      /** Prints a pattern like representation of the `Tree`.
+       *  It displays the structure of the AST.
+       */
+      def TreeCases: Printer[Tree]
```

Comment:
What `TreeStructure`?

Reply:
That was my second option but I found it a bit long. `TreeStructure`/`TypeReprStructure ` are clearly better names. I will try it and see if the code does not grow too much.

Reply:
`TreeStructure ` is long but it seems acceptable.